### PR TITLE
Use cmake::Config::get_profile() instead of stolen implementation

### PR DESCRIPTION
**Pull request by nabijaczleweli**

### CODE REVIEW

**Comment by mgeier** [comment]
Thanks, I love it when so much code can be deleted!

We should probably add the latest supported `cmake` version to `Cargo.toml`?

**Comment by nabijaczleweli** [comment]
I've changed the spec to `"^0.1.45"` which, according to https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements, should behave like `"0.1"` but with an additional lower bound on `0.1.45`

**Comment by mgeier** [comment]
Thanks!

I'm not very familiar with with this topic, but I think that `"^0.1.45"` indeed does what we want.
However, according to the paragraph just above your linked section, `"0.1.45"` should do exactly the same, since `^` seems to be the default operator.

So which should we prefer?

And if we use `^`, should we also add it to the other dependency for consistency?

**Comment by nabijaczleweli** [comment]
Personally, I'd go for `"^0.1.45"`, because I don't ever use anything but `"X.Y"` unless [strictly necessary](https://github.com/nabijaczleweli/cargo-update/blob/1c9355bc89ad6c34c183a0175832431f59f2e08d/Cargo.toml#L37-L42) – the caret stands out to me and indicates that this is an intentional axion to achieve a specific semantic and not just a copy-paste – or otherwise – error, clarified by the comment above.

But I can't speak for your conventions, it's your call.

File: Cargo.toml [hunk 1]
```diff
@@ -44,7 +44,8 @@ build-flac = ["cmake"]
 build-ogg = ["build-flac"]
 
 [build-dependencies]
-cmake = {version = "0.1", optional = true}
+# 0.1.45 adds cmake::Config::get_profile() which we need
```
Comment (mgeier) [inline Cargo.toml:47]: I think the comment is helpful, but "which we need" is redundant.
Because if we wouldn't need it, why mention it at all?

**Comment by mgeier** [comment]
I don't have any conventions yet, I'm in the process of establishing them right now!

I think I don't really care whether the caret is used, but I *do* care about consistency. This looks inconsistent to me:

```toml
cmake = "^0.1.45"
libc = "0.2"
```

It looks like the one with the caret is supposed to do something different, while in reality it does exactly the same as the one without.

I have the feeling it should be either ...

```toml
cmake = "^0.1.45"
libc = "^0.2"
```

... or ...

```toml
cmake = "0.1.45"
libc = "0.2"
```

... but not a mix of both.

Or am I missing something?

The comment is nice, but it is about the version number, it doesn't (and shouldn't?) explain the caret.

BTW, your link is different, because it uses `=`, which doesn't introduce that inconsistency.

**Comment by nabijaczleweli** [comment]
Updated

**Comment by mgeier** [comment]
Thanks a lot!

**Pull request merged**